kornia/losses/focal.py

```diff
@@ -205,16 +227,20 @@ def binary_focal_loss_with_logits(
     KORNIA_CHECK_IS_TENSOR(pos_weight)
     KORNIA_CHECK(input.shape[1] == pos_weight.shape[0], "Expected pos_weight equals number of classes.")
 
-    probs_pos = input.sigmoid()
-    probs_neg = (-input).sigmoid()
-
     log_probs_pos = nn.functional.logsigmoid(input)
     log_probs_neg = nn.functional.logsigmoid(-input)
 
-    loss_tmp = (
-        -alpha * pos_weight * probs_neg.pow(gamma) * target * log_probs_pos
-        - (1 - alpha) * probs_pos.pow(gamma) * (1.0 - target) * log_probs_neg
-    )
+    # the alpha term is not extracted to save operations
+    if alpha is None:
```

Comment:
Can you add tests for both cases?

Reply:
Sure, let me set the new design first, and more tests will be added later.